### DISCUSSION TITLE
Internal Server Error on user registration and invalid_token error when trying to login to localhost

### DISCUSSION THREAD

**Discussion by shirkan** [fief-dev/fief#158] [Q&A] [answered]
Hi, i'm trying to create a user with user/password through the login process in React.
After filling the required fields and click on "Register", I get a "Internal Server Error". 
Also, when I try to login with the user credentials, I get to the callback page, but there it is stuck. In the network tab, I see that the request to token endpoint returns with `{"error":"invalid_client"}`.

How is it possible to debug the problem? 
Running `docker logs` for the server, I see only a one-liner for the requests and response code, without too much information of what happened.

oh and one more thing if I may... how can I delete a user? how can I delete a client? is it possible to do it in the dashboard?

Many thanks.

**Comment by frankie567** [comment, marked as answer]
Several things to check:

## Do you have the latest image?

Run the following command:

```sh
docker pull ghcr.io/fief-dev/fief:latest                           
```

If an image is downloaded you should create a new container with this new image.

## Is your Client **Public**?

Browser integrations like React need the client to be **Public**: https://docs.fief.dev/configure/clients/#public-clients

## Did you correctly set `baseURL` and `clientID`?

```tsx
    <FiefAuthProvider
      baseURL="http://localhost:8000" // This should point to your worspace address
      clientId="CLIENT_ID"  // This should be your Client ID
    >
```

---

Regarding user and client deletion, this is not possible right now, but we are a [currently implementing it](https://github.com/fief-dev/fief/commit/d2c2fe1bd68d5b2a6425cee8967c335bd659009c) and it'll be part of the next version.

**Reply by shirkan** [reply]
> Several things to check:
> 
> ## Do you have the latest image?
> Run the following command:
> 
> ```shell
> docker pull ghcr.io/fief-dev/fief:latest                           
> ```
> 
> If an image is downloaded you should create a new container with this new image.
> 

Pulled new image, created new container with new image, same issues.

> ## Is your Client **Public**?
> Browser integrations like React need the client to be **Public**: https://docs.fief.dev/configure/clients/#public-clients
> 

Verified client is public.

> ## Did you correctly set `baseURL` and `clientID`?
> ```tsx
>     <FiefAuthProvider
>       baseURL="http://localhost:8000" // This should point to your worspace address
>       clientId="CLIENT_ID"  // This should be your Client ID
>     >
> ```

in my test case, the react webapp is running locally on port 3000 and the baseURL is set to the deployed one (https://fief.mydomain.com) - is this problematic? 
I added 
```
http://localhost:3000/callback
http://localhost:3000
```
to redirect URIs

> Regarding user and client deletion, this is not possible right now, but we are a [currently implementing it](https://github.com/fief-dev/fief/commit/d2c2fe1bd68d5b2a6425cee8967c335bd659009c) and it'll be part of the next version.
👍 

How would you suggest to see full debug logs from fief server (incl. exception logs)?
Thanks again.

**Reply by frankie567** [reply]
> in my test case, the react webapp is running locally on port 3000 and the baseURL is set to the deployed one (https://fief.mydomain.com/) - is this problematic?

No it should not.

> How would you suggest to see full debug logs from fief server (incl. exception logs)?

Normally, if you run `docker logs -f fief-server`, you should see all logs, including exceptions.

Are you able to access the admin dashboard correctly with your main user credentials?

**Reply by shirkan** [reply]
> Are you able to access the admin dashboard correctly with your main user credentials?

Yes I am.

**Reply by shirkan** [reply]
Mind this log from `docker logs -f fief-server`:

`INFO:     <ip>:0 - "GET /.well-known/openid-configuration HTTP/1.0" 200 OK`
`INFO:     <ip>:0 - "GET /.well-known/openid-configuration HTTP/1.0" 200 OK`
`INFO:     <ip>:0 - "POST /api/token HTTP/1.0" 200 OK`
`INFO:     <ip>:0 - "POST /api/token HTTP/1.0" 400 Bad Request`
`INFO:     <ip>:0 - "GET /.well-known/jwks.json HTTP/1.0" 200 OK`
`INFO:     <ip>:0 - "GET /.well-known/jwks.json HTTP/1.0" 200 OK`

How is it possible to see details about line #4 (400 Bad Request) ?

**Reply by frankie567** [reply]
I see that every requests are made two times. We clearly see the first `/api/token` call works, while the second does not (which is expected, an authorization code is usable only once).

I suspect it's a problem we already came across regarding [React 18 Strict Mode](https://github.com/orgs/fief-dev/discussions/92#discussioncomment-3832167). Could you try to [disable Strict Mode](https://stackoverflow.com/questions/74952506/how-to-disable-strict-mode-in-react-18) and see if it makes a difference?

**Reply by shirkan** [reply]
> I see that every requests are made two times. We clearly see the first `/api/token` call works, while the second does not (which is expected, an authorization code is usable only once).
> 
> I suspect it's a problem we already came across regarding [React 18 Strict Mode](https://github.com/orgs/fief-dev/discussions/92#discussioncomment-3832167). Could you try to [disable Strict Mode](https://stackoverflow.com/questions/74952506/how-to-disable-strict-mode-in-react-18) and see if it makes a difference?

Yes! looks like indeed after removing `React.StrictMode` element I do see it makes only 1 request which is successful. but for some reason it won't navigate back to home page as it should:

```const Callback = () => {
  const fiefAuth = useFiefAuth();
  const navigate = useNavigate();

  useEffect(() => {
    fiefAuth
      .authCallback(
        `${window.location.protocol}//${window.location.host}/callback`
      )
      .then(() => {
        navigate("/");
      })
      .catch((err) => {
        console.log(err);
      });
  }, [fiefAuth, navigate]);

  return (
    <Container>
      <h3 className="text-center">
        Logged in!
        <br />
        Returning to main page...
      </h3>
    </Container>
  );
};
```

printed error is:
```Error
    at Fief.<anonymous> (client.ts:648:1)
    at Generator.throw (<anonymous>)
    at rejected (tslib.es6.js:74:1)
```

**Reply by frankie567** [reply]
Just to be sure: you didn't enabled [ID Token encryption](https://docs.fief.dev/going-further/id-token-encryption/) on the Client, did you?

**Reply by shirkan** [reply]
> Just to be sure: you didn't enabled [ID Token encryption](https://docs.fief.dev/going-further/id-token-encryption/) on the Client, did you?

Yes, I accidentally pressed that button 😅 and have no idea how to disable it... is it possible to disable it somehow? or should I create a new client from scratch?

**Reply by frankie567** [reply]
Ha, that's it then! Well, no you'll need create a new client without encryption 😄

**Reply by shirkan** [reply]
Works perfectly now, many thanks once again 😄

**Reply by frankie567** [reply]
Ability to delete users is now released in [`v0.23.2`](https://github.com/fief-dev/fief/releases/tag/v0.23.2) and available on Fief Cloud 🚀

**Reply by shirkan** [reply]
> Ability to delete users is now released in [`v0.23.2`](https://github.com/fief-dev/fief/releases/tag/v0.23.2) and available on Fief Cloud 🚀

🥳 🥳 🥳  Updated docker image, deleting users/clients works 👍 
Thanks!
🚀 🚀 🚀
